app/controllers/TimeController.scala

```diff
@@ -0,0 +1,172 @@
+
+    for {
+      jsObjectList <- Fox.combined(futureJsObjects)
+    } yield {
+      Json.toJson(jsObjectList)
+    }
+  }
+
+  def getUsersWithWorkingHours(users: List[User], startDate: Calendar, endDate: Calendar)(implicit request: AuthenticatedRequest[AnyContent]): List[Fox[JsObject]] = {
+    users.map(user => getUserHours(user, startDate, endDate))
+  }
+
+  def getUserHours(user: User, startDate: Calendar, endDate: Calendar)(implicit request: AuthenticatedRequest[AnyContent]): Fox[JsObject] = {
+    for {
+      timeList <- TimeSpanDAO.findByUser(user, Some(startDate.getTimeInMillis), Some(endDate.getTimeInMillis))
+      timeListWithTask <- getOnlyTimeSpansWithTask(timeList)
+      timeListGreaterZero = timeListWithTask.filter(tuple => tuple._1.time > 0)
+      boxJs <- Future.traverse(timeListGreaterZero)(timeWrites(_))
+    } yield {
+      val js = boxJs.flatten
+      if(js.nonEmpty) {
+        signalOverTime(user,timeListGreaterZero)
+        js.head.value.get("time") match {
+          case Some(x) =>
+            Some(Json.obj(
+              "user" -> Json.toJson(user)(User.userCompactWrites),
+              "timelogs" -> Json.toJson(js)))
+          case _ => None
+        }
+      }
+      else
+        None
+    }
+  }
+
+  def getOnlyTimeSpansWithTask(l: List[TimeSpan])(implicit request: AuthenticatedRequest[AnyContent]): Fox[List[(TimeSpan, Task)]] = {
+    for {
+      list <- Fox.combined(l.map(t => getTimeSpanOptionTask(t)))
+    } yield {
+      list.flatten
+    }
+  }
+
+  def getTimeSpanOptionTask(t: TimeSpan)(implicit request: AuthenticatedRequest[AnyContent]): Fox[Option[(TimeSpan,Task)]] = {
+    t.annotation match {
+      case Some(annotationId) => for {
+        annotation <- AnnotationDAO.findOneById(annotationId)
+        if(annotation._task.isDefined)
+        task <- TaskService.findOneById(annotation._task.get.stringify)
+      } yield {
+        Some((t, task))
+      }
+      case None => Fox(Future(None))
+    }
+  }
+
+  def timeWrites(tuple:(TimeSpan,Task))(implicit request: AuthenticatedRequest[AnyContent]): Fox[JsObject] = {
+    for {
+      tasktype <- TaskTypeDAO.findOneById(tuple._2._taskType)
+      project <- ProjectDAO.findOneByName(tuple._2._project)
+    } yield {
+      Json.obj(
+        "time" -> formatDuration(tuple._1.time),
+        "timestamp" -> tuple._1.timestamp,
+        "annotation" -> tuple._1.annotation,
+        "_id" -> tuple._1._id.stringify,
+        "task_id" -> tuple._2.id,
+        "project_name" -> project.name,
+        "tasktype_id" -> tasktype.id,
+        "tasktype_summary" -> tasktype.summary)
+    }
+  }
+
+  def formatDuration(millis: Long): String = {
+  //example: P3Y6M4DT12H30M5S = 3 years + 9 month + 4 days + 12 hours + 30 min + 5 sec
+  // only hours, min and sec are important in this scenario
+    val h = (millis / 3600000)
+    val m = (millis / 60000) % 60
+    val s = (millis / 1000) % 60
+
+    var res = "PT"
```

Comment:
Is there anything wrong with having `0`s in the formatted duration? Otherwise I would suggest getting rid of the conditions here and simply use `s"PT${h}H${m}M${s}S"`